### PR TITLE
avoid null ref exception on showing upgrade to package ref menu

### PR DESCRIPTION
**Pull request by rohit21agrawal**
Fix VSTS internal issue : 619387

Since there is no CAB available, trying to avoid null ref based on guesswork.

### CODE REVIEW

**Comment by rohit21agrawal** [comment]
🔔

**Review by dtivel — COMMENTED** [review]
It looks like the problem could exist elsewhere in the file ([example](https://github.com/NuGet/NuGet.Client/pull/2317/files#diff-d872f9d82134c587ae15bdcec1a2e823L850)).  Can you guard against `ConsoleStatus` being `null` in all locations?

You can create a helper method to simplify call sites.

File: src/NuGet.Clients/NuGet.Tools/NuGetPackage.cs [hunk 1]
```diff
@@ -826,9 +826,14 @@ private void BeforeQueryStatusForUpgradeNuGetProject(object sender, EventArgs ar
...
-
+                var isConsoleBusy = false;
+                if (ConsoleStatus != null)
+                {
+                    isConsoleBusy = ConsoleStatus.Value.IsBusy;
+                }
+                
                 command.Visible = IsSolutionOpen && await IsProjectUpgradeableAsync();
-                command.Enabled = !ConsoleStatus.Value.IsBusy && IsSolutionExistsAndNotDebuggingAndNotBuilding() && HasActiveLoadedSupportedProject;
+                command.Enabled = isConsoleBusy && IsSolutionExistsAndNotDebuggingAndNotBuilding() && HasActiveLoadedSupportedProject;
```
Comment (jainaashish) [inline src/NuGet.Clients/NuGet.Tools/NuGetPackage.cs:839]: IMO you can just add a null condition for ConsoleStatus and skip doing anything if that's null. 

And also add the same condition in `BeforeQueryStatusForUpgradePackagesConfig` method.

**Comment by rohit21agrawal** [comment]
@dtivel @jainaashish updated.

File: src/NuGet.Clients/NuGet.Tools/NuGetPackage.cs [hunk 2]
```diff
@@ -806,11 +806,14 @@ private void BeforeQueryStatusForPowerConsole(object sender, EventArgs args)
                 NuGetUIThreadHelper.JoinableTaskFactory.Run(InitializeMEFAsync);
             }
 
+            var isConsoleBusy = false;
             if (ConsoleStatus != null)
             {
-                var command = (OleMenuCommand)sender;
-                command.Enabled = !ConsoleStatus.Value.IsBusy && !_powerConsoleCommandExecuting;
+                isConsoleBusy = ConsoleStatus.Value.IsBusy;
```
Comment (dtivel) [inline src/NuGet.Clients/NuGet.Tools/NuGetPackage.cs:812]: You could dedupe a lot of code you're adding in this PR with a helper property (`IsConsoleBusy`) or helper method (`GetIsConsoleBusy()`).

**Review by dtivel — APPROVED** [review]

File: src/NuGet.Clients/NuGet.Tools/NuGetPackage.cs [hunk 3]
```diff
@@ -806,11 +806,14 @@ private void BeforeQueryStatusForPowerConsole(object sender, EventArgs args)
...
             }
+
+            var command = (OleMenuCommand)sender;
+            command.Enabled = isConsoleBusy && !_powerConsoleCommandExecuting;
```
Comment (jainaashish) [inline src/NuGet.Clients/NuGet.Tools/NuGetPackage.cs:816]: This is not correct, you reversed the condition. If console is not busy then you want to enable command, not the other way around.

**Comment by rohit21agrawal** [comment]
yeah, i realized a few mins back. sending out a fix soon.

File: src/NuGet.Clients/NuGet.Tools/NuGetPackage.cs (hunk 3 above)
Comment (rohit21agrawal) [reply src/NuGet.Clients/NuGet.Tools/NuGetPackage.cs:816]: yeah, i realized a few mins back. sending out a fix soon.

**Pull request merged**